nixos/modules/services/desktops/flatpak.nix

```diff
@@ -43,6 +100,11 @@ in {
       "/var/lib/flatpak/exports"
     ];
 
+    system.activationScripts.flatpak-setup = ''
```

Comment:
I've used this before in nix-flatpak and it always failed at startup due to not having internet when using NetworkManager. What is being done to avoid this here?

Reply:
Yes you're true, although I haven't noticed this issue before i think it will be a good idea adding a systemd service definition to prevent this issue from happening?
I think something like this will work.
```nix
systemd.services.flatpak-setup = {
  description = "Flatpak package management";
  wantedBy = [ "multi-user.target" ];
  after = [ "network-online.target" ];
  wants = [ "network-online.target" ];
  path = [ pkgs.flatpak ];
  script = ''
    ${manageFlatpaks}
  '';
  serviceConfig = {
    Type = "oneshot";
    RemainAfterExit = true;
  };
};
```

Reply:
I still kept the activation script because they serve different purposes. The activation script is for installation and optional package updates during build and the systemd service is for automatic updates (in a specific time period).